### DISCUSSION TITLE
Settings: Authentication: Login form is deactivated but is still shown when Google SAML is activated

### DISCUSSION THREAD

**Discussion by EduNetEurope** [kimai/kimai#4725] [Questions & Answers] [answered]
We have successfully activated Google SAML.
We tried to [hide the login form](https://www.kimai.org/documentation/configurations.html#authentication), but despite the setting, the login form is still there.
Will it only disappear once ALL users have been [upgraded to SAML login](https://www.kimai.org/documentation/cloud-saml.html#important-to-know)?

**Comment by kevinpapst** [comment]
<img width="267" alt="Bildschirmfoto 2024-04-05 um 00 14 11" src="https://github.com/kimai/kimai/assets/533162/43c333e1-5ae8-4095-8ad5-dc7f1ec848f5">
<img width="541" alt="Bildschirmfoto 2024-04-05 um 00 15 19" src="https://github.com/kimai/kimai/assets/533162/d813fc52-71c5-4c62-99bd-22ae5d0a2a02">

> Will it only disappear once ALL users have been [upgraded to SAML login](https://www.kimai.org/documentation/cloud-saml.html#important-to-know)?

No

**Comment by EduNetEurope** [comment]
Setting:
![image](https://github.com/kimai/kimai/assets/148647197/702cfd82-0bc0-4b6f-8130-48032c99fece)

Login Page:
![image](https://github.com/kimai/kimai/assets/148647197/8e0a19cc-7f99-40d1-a261-51bb7fd9880c)


The setting is ignored.

**Reply by kevinpapst** [reply]
local.yaml can overwrite it

**Reply by EduNetEurope** [reply]
We have added the Googlse SAML settings from [here](https://www.kimai.org/documentation/saml-google.html) and we have not adjusted anything else (except: we set "resetOnLogin" to "false"). 
The authentication setting showed up after activating SAML, but it does not do anything.

So I suppose we need to manually add to local.yaml that the form should be deactivated.

Unfortunately, we cannot find the necessary piece of code. 
This is where we looked for it:

- https://www.kimai.org/documentation/local-yaml.html
- https://github.com/kimai/kimai/blob/main/config/packages/kimai.yaml

What do we have to add to local.yaml to hide the login form?

**Comment by EduNetEurope** [comment]
Can anyone help us out here? I suppose we probably only need to add one line of code to local.yaml.

**Comment by EduNetEurope** [comment, marked as answer]
Should anyone face the same issue, this is the solution.
Download and install [CustomCSS plugin](https://www.kimai.org/store/keleo-css-custom-bundle.html).

Add the following code:

/* START [login] hide login email form */
body.login-page form.login-box-body.security-login  { display: none; }
/* END  */

/* START [login] hide "oder" */
body.login-page .hr-text  { display: none; }
/* END */

/* START [login] hide "Anmelden" */
body.login-page .card-title  { display: none; }
/* END */

/* START [login] center vertically Google Login */
body.login-page .card-body  { padding-top:0; }
/* END */

/* START [login] background color: white */
body.login-page { background: #ffffff; }
/* END  */

/* START [login] hide grey box border */
body.login-page .card.card-md { border-color: #ffffff; }
/* END  */

**Reply by kevinpapst** [reply]
Without seeing the local.yaml and the system and the versions, I cannot tell you, why the login form is shown. 
But this here is a hack and NOT the solution.
The solution is to use the setting, which works as expected.

**Comment by EduNetEurope** [comment]
This is our complete local.yaml.

We copied the entire SAML configuration from [here](https://www.kimai.org/documentation/saml.html) and adjusted the keys outlined [here](https://www.kimai.org/documentation/saml-google.html).

The only other changes we made:
- We deleted the mapping part.
- We set resetOnLogin to false.

We run Kimai 2.21.0 (installed through Cloudron).
SAML was activated in v. 2.9.4 on 26 March.

`# See https://www.kimai.org/documentation/local-yaml.html

kimai:
    saml:
        provider: google
        activate: true
        title: Login with Google
        mapping:
            - { saml: $Email, kimai: email }
            - { saml: $FirstName $LastName, kimai: alias }
        roles:
            resetOnLogin: false
            attribute: Groups
           
                
        connection:
            idp:
                entityId: 'https://accounts.google.com/o/saml2?idpid=XXX'
                singleSignOnService:
                    url: 'https://accounts.google.com/o/saml2/idp?idpid=XXX'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                #singleLogoutService:
                #    url: 'https://www.example.com/logout'
                #    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                x509cert:   'XXX'
            # Your Kimai: replace https://XXX with your base URL
            sp:
                entityId: 'https://XXX/auth/saml/metadata'
                assertionConsumerService:
                    url: 'https://XXX/auth/saml/acs'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST'
                singleLogoutService:
                    url: 'https://XXX/auth/saml/logout'
                    binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'
                #privateKey: ''
            # only set baseurl, if auto-detection doesn't work
            #baseurl: ''
            strict: true
            debug: true
            security:
                nameIdEncrypted: false
                authnRequestsSigned: false
                logoutRequestSigned: false
                logoutResponseSigned: false
                wantMessagesSigned: false
                wantAssertionsSigned: false
                wantNameIdEncrypted: false
                requestedAuthnContext: true
                signMetadata: false
                wantXMLValidation: true
                signatureAlgorithm: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'
                digestAlgorithm: 'http://www.w3.org/2001/04/xmlenc#sha256'
            contactPerson:
                technical:
                    givenName: 'XXX'
                    emailAddress: 'XXX'
                support:
                    givenName: 'XXX'
                    emailAddress: 'XXX'
            organization:
                en:
                    name: 'XXX'
                    displayname: 'XXX'
                    url: 'XXX'

 
`
